pkg/apis/azureprovider/v1alpha1/types.go

```diff
@@ -369,10 +369,11 @@ type VM struct {
 }
 
 type Image struct {
-	Publisher string `json:"publisher"`
-	Offer     string `json:"offer"`
-	SKU       string `json:"sku"`
-	Version   string `json:"version"`
+	Publisher  string `json:"publisher"`
+	Offer      string `json:"offer"`
+	SKU        string `json:"sku"`
+	Version    string `json:"version"`
+	ResourceID string `json:"resourceID"`
```

Comment:
how is this related to this PR/commit?

Reply:
+1, it does not seem the new field has anything to do with allowing to inject user defined data. Can you open a separate PR for it? Resp. does it need to be implemented before the user data can be constructed?

Reply:
Agreed doesnt have to be part of the PR, fixing